src/renderers/shared/fiber/ReactFiber.js

```diff
@@ -163,59 +163,54 @@ if (__DEV__) {
   var debugCounter = 1;
 }
 
-// This is a constructor of a POJO instead of a constructor function for a few
-// reasons:
+function FiberNode(tag: TypeOfWork, key: null | string) {
```

Comment:
It's a bit unfortunate that we can't use the `Fiber` name because it conflicts with the type. However we could if we switched this to use the class syntax. We just have to ensure that our class syntax transform doesn't add the extra checks that babel does by default.